kolibri_instant_schools_plugin/assets/src/views/sign-up-page/index.vue

```diff
@@ -87,10 +86,37 @@
         @input="updateSelection"
       />
 
+      <!-- ToS modal only shows up if the box empty prior to clicking -->
+      <!-- Using UI Checkbox for blur events and v-model -->
+      <ui-checkbox
```

Comment:
We should use the `k-checkbox`. Just emit a new blur event within the k-checkbox. Also, you can do what you are doing with the `v-model` and `@change` by just having something like `@change="handleCheckboxChange"`

```javascript
handleCheckboxChange(newValue) {
  this.termsAgreed = newValue;
  this.showTerms = this.termsAgreed; 
},
```

Reply:
Wasn't sure if bringing in more changes to the kolibri codebase is worth it. I had something similar to that block ^ before I added validation.

Reply:
Well we definitely do not want to use the ui-checkbox, since that kind of defeats the purpose of having created our own in-house checkbox component. :cry: